File: scala/src/main/scala/intro/Complex.scala
package intro

class Complex(real: Double, imaginary: Double) {
  def re = real
  def im = imaginary

  override def toString: String = re.toString + (if (im > 0) s" + $im" + "i" else "")
}

object ComplexTest {
  def main(args: Array[String]): Unit = {
    val c = new Complex(4,0.5)

    println(c.toString)
  }
}

File: src/main/java/org/geeksforgeeks/practice/basics/Complex.java
package org.geeksforgeeks.practice.basics;

public class Complex {
    int real;
    int comp;

    Complex(int r, int c) {
        real = r;
        comp = c;
    }

    Complex() {
        this(0, 0);
    }

    /**
     * @param c Complex object
     * @return Sum of this complex number with argument.
     */
    public Complex add(Complex c) {
        this.real += c.real;
        this.comp += c.comp;

        return this;
    }

    @Override
    public String toString() {
        if (comp >= 0) {
            return (real + " + i" + comp);
        } else {
            return (real + " - i" + Math.abs(comp));
        }
    }
}
